src/05-run.md

```diff
@@ -0,0 +1,164 @@
+# Advanced Overview of `cosock.run`
```

Comment:
This goes into a lot of internal detail. Thoughts on moving it to an appendix/late section dedicated to internal details? Just to make it clear to users that they don't need to understand it to use the library.